Tribler/Core/Modules/resource_monitor.py

```diff
@@ -62,6 +66,19 @@ def check_resources(self):
         elif hasattr(self.process, "memory_info") and callable(getattr(self.process, "memory_info")):
             self.memory_data.append((time_seconds, self.process.memory_info().rss))
 
+        # Check for available disk space
+        disk_usage = psutil.disk_usage(".")
```

Comment:
Is this the same as the disk being downloaded to?

Reply:
Interesting question. Insufficient disk space when downloading torrents should be handled by libtorrent already IIRC. We're writing quite some stuff to our Tribler state dir (`.Tribler`) and theoretically, this could be on another disk. This code line fetches the available space of the disk where the _Python source files_ are located, which might be a different disk than the state directory.

TLDR: I would change it to something like `psutil.disk_usage(self.session.config.get_state_dir())`.